integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFormatsIT.java

```diff
@@ -120,6 +122,31 @@ public void testNumericFormats() {
         rows("1970-01-02 03:55:00", "1970-01-01 00:01:40.5"));
   }
 
+  @Test
+  @SneakyThrows
+  public void testDateNanosWithFormats() {
```

Comment:
Maybe add a test that uses custom format with fractions of a second?

Reply:
See next test (line 140)